### DISCUSSION TITLE
exchangelib.errors.ErrorFolderNotFound: No subfolder with name 'SUBFOLDER NAME'

### DISCUSSION THREAD

**Discussion by Tre-Seibert** [ecederstrand/exchangelib#1286] [Q&A] [answered]
I have a web application built on exchangelib that retrieves task items. The app runs 24/7 and has worked great since august of 2023. As of this week the web app suddenly started encountering this error:

"    raise ErrorFolderNotFound(f"No subfolder with name {other!r}")
exchangelib.errors.ErrorFolderNotFound: No subfolder with name 'SUBFOLDER NAME'"

The code hasn't changed in months, neither has the folder "SUBFOLDER NAME" or any of the properties associated with the folder. The folder is in my public folders root. I also haven't updated exchangelib since August, it is currently running exchangelib version 5.0.3. Any ideas as to why this is happening? 

Additionally, after printing out child_folder_count for my public folders root it confirms that there are 2 child folders which can be found in the picture commented below. I've also tried accessing the other folder, "SUBFOLDER NAME2" and I am given the same error:

"    raise ErrorFolderNotFound(f"No subfolder with name {other!r}")
exchangelib.errors.ErrorFolderNotFound: No subfolder with name 'SUBFOLDER NAME2'"

When I try printing out the tree of my public folders using "print(account.public_folders_root.tree())" I get this output:
"IPM_SUBTREE"

Here is the XML Logs:

Request XML:
```xml
<?xml version='1.0' encoding='utf-8'?>
<s:Envelope
    xmlns:s="http://schemas.xmlsoap.org/soap/envelope/"
    xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"
    xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
  <s:Header>
    <t:RequestServerVersion Version="Exchange2016"/>
    <t:TimeZoneContext>
      <t:TimeZoneDefinition Id="Eastern Standard Time"/>
    </t:TimeZoneContext>
  </s:Header>
  <s:Body>
    <m:GetFolder>
      <m:FolderShape>
        <t:BaseShape>IdOnly</t:BaseShape>
        <t:AdditionalProperties>
          <t:FieldURI FieldURI="folder:DisplayName"/>
          <t:FieldURI FieldURI="folder:FolderClass"/>
          <t:FieldURI FieldURI="folder:PermissionSet"/>
        </t:AdditionalProperties>
      </m:FolderShape>
      <m:FolderIds>
        <t:FolderId Id="" ChangeKey=""/>
        <t:FolderId Id="" ChangeKey=""/>
      </m:FolderIds>
    </m:GetFolder>
  </s:Body>
</s:Envelope>
```

Response XML:
```xml
<?xml version='1.0' encoding='utf-8'?>
<s:Envelope
    xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
  <s:Header>
    <h:ServerVersionInfo
    xmlns:h="http://schemas.microsoft.com/exchange/services/2006/types"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" MajorVersion="15" MinorVersion="20" MajorBuildNumber="7409" MinorBuildNumber="31" Version="V2018_01_08"/>
  </s:Header>
  <s:Body>
    <m:GetFolderResponse
    xmlns:m="http://schemas.microsoft.com/exchange/services/2006/messages"
    xmlns:xsd="http://www.w3.org/2001/XMLSchema"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns:t="http://schemas.microsoft.com/exchange/services/2006/types">
      <m:ResponseMessages>
        <m:GetFolderResponseMessage ResponseClass="Success">
          <m:ResponseCode>NoError</m:ResponseCode>
          <m:Folders>
            <t:Folder>
              <t:FolderId Id="" ChangeKey=""/>
              <t:FolderClass>IPF.Note</t:FolderClass>
              <t:DisplayName>SUBFOLDER NAME2</t:DisplayName>
            </t:Folder>
          </m:Folders>
        </m:GetFolderResponseMessage>
        <m:GetFolderResponseMessage ResponseClass="Success">
          <m:ResponseCode>NoError</m:ResponseCode>
          <m:Folders>
            <t:Folder>
              <t:FolderId Id="" ChangeKey=""/>
              <t:DisplayName>SUBFOLDER NAME</t:DisplayName>
            </t:Folder>
          </m:Folders>
        </m:GetFolderResponseMessage>
      </m:ResponseMessages>
    </m:GetFolderResponse>
  </s:Body>
</s:Envelope>
```

```python
Traceback (most recent call last):
  File "", line 250, in home
    fParent = fPublic / fTB
              ~~~~~~~~^~~~~
  File "/usr/local/lib/python3.11/site-packages/exchangelib/folders/base.py", line 783, in __truediv__
    raise ErrorFolderNotFound(f"No subfolder with name {other!r}")
exchangelib.errors.ErrorFolderNotFound: No subfolder with name 'SUBFOLDER NAME'
```

**Comment by ecederstrand** [comment]
The response XML indicates that the subfolder in question *is* returned (plus another one with almost the same name). I would go into exchangelib/folders/base.py line 783 and add a `breakpoint()` to see why we are falling through to where `ErrorFolderNotFound` is raised instead of returning the child folder in question.

**Reply by Tre-Seibert** [reply]
This is a high priority bug that I need to fix. Is there a way to reference the folder using the folder ID instead of the folder name to see if that works to return the folder?

**Comment by Tre-Seibert** [comment]
Honestly due to my incompetence with python debugger I am not too sure what to check for once we get to this breakpoint to further debug.

**Comment by Wonkyung-LEE** [comment]
i have same problem. [here](https://github.com/ecederstrand/exchangelib/issues/1273#issuecomment-2026632936)
cause maybe response "folder-id" not in "_folders_map".
so, [get_folder] method in "root.py" always return None, then parent id always has None value.
anybody can help this issue?

**Reply by Tre-Seibert** [reply]
Im sorry I still havent found a solution

**Comment by Tre-Seibert** [comment, marked as answer]
This issue is related to #1273 . A temporary workaround to this issue can be used as follows until a exchangelib update is released to address this issue:

```
fav_folder = None
inbox_folders = FolderCollection(account=account, folders=[account.inbox])
for folder in inbox_folders.find_folders(depth="Shallow"):
    if folder.name == "MyFavouriteSubFolder":
        fav_folder = folder
        
for mail in fav_folder.all().only("subject", "sender", "datetime_received"):
    print(mail.sender.email_address)

```
